query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala

```diff
@@ -59,15 +71,34 @@ class CountOverTimeFunction extends RangeFunction {
   override def apply(startTimestamp: Long, endTimestamp: Long, window: Window,
                      sampleToEmit: TransientRow,
                      queryConfig: QueryConfig): Unit = {
-    sampleToEmit.setValues(endTimestamp, count)
+    sampleToEmit.setValues(endTimestamp, count.toDouble)
   }
 }
 
-class AvgOverTimeFunction extends RangeFunction {
-
-  var sum = 0d
-  var count = 0
+class CountOverTimeChunkedFunction(var count: Int = 0) extends ChunkedRangeFunction {
+  override final def reset(): Unit = { count = 0 }
+  def apply(endTimestamp: Long, sampleToEmit: TransientRow): Unit = {
+    sampleToEmit.setValues(endTimestamp, count.toDouble)
+  }
+  def addChunks(tsCol: Int, valueCol: Int, info: ChunkSetInfo, startTime: Long, endTime: Long): Unit = {
+    val timestampVector = info.vectorPtr(tsCol)
+    val tsReader = bv.LongBinaryVector(timestampVector)
+
+    // First row >= startTime, so we can just drop bit 31 (dont care if it matches exactly)
+    val startRowNum = tsReader.binarySearch(timestampVector, startTime) & 0x7fffffff
+    val endRowNum = tsReader.binarySearch(timestampVector, endTime) match {
+      // if endTime does not match, we want last row such that timestamp < endTime
+      // Note if we go past end of timestamps, it will never match, so this should make sure we don't go too far
+      case row if row < 0 => (row & 0x7fffffff) - 1
```

Comment:
If row is 0x80000000, then this yields -1.

Reply:
Yes, exactly.  That's the intended result, it indicates there is no such row satisfying the ceiling time constraint.